skimage/morphology/_watershed.pyx

```diff
@@ -10,87 +10,115 @@ All rights reserved.
 Original author: Lee Kamentsky
 """
 import numpy as np
-cimport numpy as np
+cimport numpy as cnp
 cimport cython
 
 
-ctypedef np.int32_t DTYPE_INT32_t
+ctypedef cnp.int32_t DTYPE_INT32_t
 DTYPE_BOOL = np.bool
-ctypedef np.int8_t DTYPE_BOOL_t
+ctypedef cnp.int8_t DTYPE_BOOL_t
 
 
 include "heap_watershed.pxi"
 
 
+cdef extern from "math.h":
```

Comment:
@ahojnnes I had this initially but it actually resulted in way more code generated for the sqrt call! Perusing the Cython docs suggested this. I don't understand why this was an issue.